core/instance.go

```diff
@@ -504,6 +497,20 @@ func (i *instance) generateTagsList() []*ec2.TagSpecification {
 		},
 	}
 
+	if i.asg.LaunchTemplate != nil {
+		tags.Tags = append(tags.Tags, &ec2.Tag{
+			Key:   aws.String("LaunchTemplateName"),
+			Value: i.asg.LaunchTemplate.LaunchTemplateName,
+		})
+	}
+
+	if i.asg.LaunchConfigurationName != nil {
+		tags.Tags = append(tags.Tags, &ec2.Tag{
+			Key:   aws.String("LaunchConfigurationName"),
+			Value: i.asg.LaunchConfigurationName,
+		})
+	}
```

Comment:
Logic wise wouldn't that be an `if / else if` - rather than 2 if?

Reply:
AWS will make sure an ASG can only have one or the other, so the if/else or if/if will act the same, from what I have seen. That being said, if you prefer if/else I can fix the PR?